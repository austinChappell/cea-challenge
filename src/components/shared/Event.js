import React from 'react';
import moment from 'moment';

import Card from './Card';

const Event = (props) => {
  const { event } = props;
  const {
    rsvp_limit: rsvpLimit,
    name,
    time,
    yes_rsvp_count: rsvp,
  } = event;
  return (
    <Card>
      <div className="Event">
        <h4>{name}</h4>
        <h6>{moment(time).format('MMMM Do YYYY, h:mm a')}</h6>
        <h6>{rsvp} going</h6>
        <h6>{rsvpLimit - rsvp} spots remaining</h6>
        <button>
          Learn More
        </button>
      </div>
    </Card>
  );
};

export default Event;
