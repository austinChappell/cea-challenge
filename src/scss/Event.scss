.Event {
  h4 {
    color: $color-secondary-dark;
    margin-bottom: 10px;
  }
  h6 {
    margin-bottom: 10px;
  }
  button {
    background-color: $color-primary;
    border: none;
    border-radius: $standard-border-radius;
    color: $color-white;
    cursor: pointer;
    outline: none;
    padding: 4px 8px;
    font-size: 1rem;
    transition: 500ms;
    &:hover {
      background-color: $color-primary-light;
    }
  }
}