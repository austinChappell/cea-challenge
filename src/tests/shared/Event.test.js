import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Event from '../../components/shared/Event';

const props = {
  event: {
    rsvp_limit: 40,
    name: 'ReactJs Dallas',
    time: 1522030848733,
    yes_rsvp_count: 40,
  },
  selectEvent: () => {},
};

describe('The Event Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Event {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
