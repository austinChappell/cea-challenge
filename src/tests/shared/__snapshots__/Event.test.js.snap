// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Event Component should render correctly 1`] = `
<Card
  handleClick={null}
  returnValue={null}
>
  <div
    className="Event"
  >
    <h4>
      ReactJs Dallas
    </h4>
    <h6>
      March 25th 2018, 9:20 pm
    </h6>
    <h6>
      40
       going
    </h6>
    <h6>
      0
       spots remaining
    </h6>
    <button
      onClick={[Function]}
    >
      Learn More
    </button>
  </div>
</Card>
`;
